clang/include/clang/Basic/Attr.td

```diff
@@ -1435,6 +1435,38 @@ def CXX11NoReturn : InheritableAttr {
   let Documentation = [CXX11NoReturnDocs];
 }
 
+def NonBlocking : TypeAttr {
+  let Spellings = [CXX11<"clang", "nonblocking">,
```

Comment:
I don't really care either way, but there was a comment suggesting namespacing the GNU-style attribute, just in case GNU were to add their own attribute with the same name.

There is also precedent: `clang_builtin_alias`

Reply:
CC @erichkeane 

`clang_builtin_alias` is the only attribute we prefix in that way; I don't think it's a good precedent. GNU attributes are a shared namespace, but we have plenty of other examples where we define a GNU attribute that GCC doesn't support (overloadable, noderef, preferred_name, etc).

Reply:
I agree with Aaron here, doing that sort of prefix is awkward.  As Aaron says, the GNU attributes are a shared namespace, and we are both typically mutually respectful of that.  I'd hope that if we get implementation that GCC would use it to implement the same functionality.

It is also specific enough that I don't see it being used for anything else.

Additionally, I DISAGREE with past Aaron that asked for the `clang_builtin_alias` spelling.  IMO, that was not well motivated, different from what we've done historically, and frankly, a mistake.

Reply:
Yeah, Present Aaron regrets Past Aaron's caution there (there was a concern that `builtin_alias` really was plausibly going to be used by GCC but with a potentially different syntax, but that's the case with all attributes in the shared namespace, so I really shouldn't have asked for that prefix. Mea culpa!).

Reply:
I've gotten rid of the "clang_" prefix on the GNU-style attributes.

Reply:
Present Aaron thanks you and Past Aaron apologized for the churn. ;-)